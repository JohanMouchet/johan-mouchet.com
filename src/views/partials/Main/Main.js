// @flow

import * as React from "react";
import cx from "classnames";
import "./Main.scss";

type Props = {
  margin?: "no-margin",
  className?: string | Array<string> | Object,
  children: Node,
};

const Main = ({ margin, className, children }: Props) => {
  const classNames = cx("p-main", margin && `p-main--${margin}`, className);

  return <main className={classNames}>{children}</main>;
};

export default Main;
