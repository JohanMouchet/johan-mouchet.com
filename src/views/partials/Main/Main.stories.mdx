import { Canvas, Meta, Story } from "@storybook/addon-docs";
import { select, text } from "@storybook/addon-knobs";
import Main from ".";

<Meta title="Partials/Main" component={Main} />

# Main

<Canvas>
  <Story name="default">
    <div style={{ outline: "1px dotted" }}>
      <Main
        margin={select("margin", {
          "": null,
          "no-margin": "no-margin",
        })}
      >
        {text("children", "children")}
      </Main>
    </div>
  </Story>
</Canvas>
