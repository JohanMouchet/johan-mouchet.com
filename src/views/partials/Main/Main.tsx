import * as React from "react";
import cx from "classnames";
import "./Main.scss";

type Props = {
  margin?: "no-margin";
  className?: string | string[] | { [key: string]: boolean };
  children: React.ReactNode;
};

const Main: React.FC<Props> = ({ margin, className, children }) => {
  const classNames = cx("p-main", margin && `p-main--${margin}`, className);

  return <main className={classNames}>{children}</main>;
};

export default Main;
