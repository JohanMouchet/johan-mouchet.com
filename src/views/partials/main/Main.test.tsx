import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import { Main } from "./Main";

it("Renders <Main /> without crashing", () => {
  expect(render(<Main>children</Main>).container.firstChild).toMatchSnapshot();
});
