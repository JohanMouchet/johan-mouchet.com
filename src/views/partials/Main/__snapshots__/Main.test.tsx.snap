// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Main /> without crashing 1`] = `
<main
  className="p-main"
/>
`;
